components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -807,7 +807,7 @@ public Subscriber getSubscriber(int subscriberId) throws APIManagementException
         return null;
     }
 
-    public int addSubscription(ApiTypeWrapper apiTypeWrapper, int applicationId, String status)
+    public int addSubscription(ApiTypeWrapper apiTypeWrapper, int applicationId, String status, String subscriber)
```

Comment:
subscriber parameter defined but not used. I think you need to insert the value into DB when adding the subscription to fix the bug.

Reply:
https://github.com/wso2/carbon-apimgt/pull/8531/commits/32efd32edd3c2ca791281eea000f281341d39bd6